### PR TITLE
Add IT for pipeline with MSBuild

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: its/src/test/java/com/sonar/it/jenkins/orchestrator/JenkinsOrchestrator.java [hunk 1]
```diff
@@ -267,14 +270,9 @@ public JenkinsOrchestrator newFreestyleJobWithMaven(String jobName, File project
...
-    
+
     findElement(By.linkText("Invoke top-level Maven targets")).click();
-    try {
-      Thread.sleep(2000);
-    } catch (InterruptedException e1) {
-      e1.printStackTrace();
-    }
-    setTextValue(findElement(By.name("_.targets"), 1), getMavenParams(orchestrator));
+    setTextValue(findElement(driver, By.name("_.targets"), 1), getMavenParams(orchestrator));
```
Comment (janos-ss) [inline its/src/test/java/com/sonar/it/jenkins/orchestrator/JenkinsOrchestrator.java]: FYI, the sleep was added as a dirty workaround for a timing issue in ITs executed by cix. Without this sleep, a window will pop-up asking if you want to navigate away from the page with unsaved changes. That said, the sleep was not intended as a solution, just an experiment to see if it indeed helps. It did seem to help, but then we didn't clean it up. If you remove this, probably the build will fail on cix, complaining about a pop-up.

File: src/main/java/hudson/plugins/sonar/AbstractMsBuildSQRunner.java [hunk 2]
```diff
@@ -108,24 +76,23 @@ protected String getExeName(MsBuildSQRunnerInstallation msBuildScanner, EnvVars
...
-  protected static void saveScannerName(Run<?, ?> r, @Nullable String name) throws IOException, InterruptedException {
-    r.addAction(new InjectEnvVarAction(Collections.singletonMap(INST_NAME_KEY, Util.fixNull(name))));
-  }
+  static class SonarQubeScannerMsBuildParams extends InvisibleAction {
 
-  protected static void saveSonarInstanceName(Run<?, ?> r, @Nullable String name) throws IOException, InterruptedException {
-    r.addAction(new InjectEnvVarAction(Collections.singletonMap(SONAR_INST_NAME_KEY, Util.fixNull(name))));
-  }
+    private String scannerName;
+    private String sqServerName;
```
Comment (janos-ss) [inline src/main/java/hudson/plugins/sonar/AbstractMsBuildSQRunner.java]: Can't these be `final` ?

Comment (henryju) [inline src/main/java/hudson/plugins/sonar/AbstractMsBuildSQRunner.java]: Indeed

File: its/src/test/java/com/sonar/it/jenkins/orchestrator/JenkinsOrchestrator.java (hunk 1 above)
Comment (henryju) [inline its/src/test/java/com/sonar/it/jenkins/orchestrator/JenkinsOrchestrator.java]: ITs are green on my branch on cix so I think the workaround is no more needed

**Pull request merged**